apps/api/lib/helpers/withMiddleware.ts

```diff
@@ -41,7 +44,8 @@ const withMiddleware = label(
     "verifyApiKey",
     "rateLimitApiKey",
     "addRequestId",
-  ] // <-- Provide a list of middleware to call automatically
-);
+  ] as Middleware[]; // <-- Provide a list of middleware to call automatically
+
+const withMiddleware = label(middleware, middlewareOrder);
 
-export { withMiddleware };
+export { withMiddleware, middleware, middlewareOrder };
```

Comment:
Do we actually need to export the `middlewareOrder`? Are we not able to take the `withMiddleware` object and assert things about it in the tests to ensure the order is correct?

Reply:
@sean-brydon and I talked about this and it's not straightforward. We are sticking with this approach for now.